pkg/psmdb/mongo/mongo.go

```diff
@@ -39,17 +39,12 @@ func Dial(addrs []string, replset, username, password string, useTLS bool) (*mon
 		return nil, fmt.Errorf("failed to connect to mongo rs: %v", err)
 	}
 
-	defer func() {
-		if err != nil {
-			_ = client.Disconnect(ctx)
-		}
-	}()
-
```

Comment:
Why it was removed?

Reply:
to simplify code
Now instead of this statement, we have just `client.Disconnect(ctx)` in case if we have error from ping

Reply:
But.
1. Err is not checked
2. If you add more code after ping which also  can return error you will have to do this disconnect again.
So please leave defer

Reply:
But.
1. Before error is also not checked(you are using `_ = client.Disconnect(ctx)`
2. No more code should be added, if we can't reach database. If we continue execution, we should not close connection
In the end, it's `dial` function, which should only establish connection

Reply:
It's fixed here https://github.com/percona/percona-server-mongodb-operator/pull/435

Reply:
fixed